src/endpoints/usernames/username.controller.ts

```diff
@@ -16,12 +16,12 @@ export class UsernameController {
     status: 404,
     description: 'Username not found'
   })
-  async getUsernameDetails(@Param('username') username: string, @Res() res: any): Promise<any> {
+  async getUsernameDetails(@Param('username') username: string): Promise<{ address: string }> {
     let address = await this.usernameService.getUsernameAddress(username);
     if (!address) {
       throw new HttpException('Account not found', HttpStatus.NOT_FOUND);
     }
 
-    return res.redirect(`/accounts/${address}`);
+    return { address };
```

Comment:
In this case request need to be redirected to `/accounts/address`, not to return the username address